app/src/main/java/com/chess/clock/engine/CountDownTimer.java

```diff
@@ -55,18 +55,17 @@ public void run() {
 			*/
             if (mLastTickTime > 0) {
                 long elapsedTime = System.currentTimeMillis() - mLastTickTime;
-                mTime -= elapsedTime;
+                setTime(mTime - elapsedTime);
             } else {
                 // When last tick time is not available, update using fixed count down interval.
-                mTime -= mCountDownInterval;
+                setTime(mTime - mCountDownInterval);
             }
 
             // Store current tick time
             mLastTickTime = System.currentTimeMillis();
```

Comment:
If we want to avoid clocks jumping on system clock change, there is also `SystemClock.elapsedRealtime()`.
It's monotonic and could be fine to user for our purpose, unless we want to survive System restart (it resets on boot).

Reply:
Sure, will update 👍